Octokit.Tests.Integration/Clients/RepositoryBranchesClientTests.cs

```diff
@@ -543,4 +544,999 @@ public TheDeleteBranchProtectionMethod()
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+            var contextOrgTeam2 = await _github.CreateTeamContext(Helper.Organization, new NewTeam(Helper.MakeNameWithTimestamp("team2")));
+
+            // Grant team push access to repo
+            await _github.Organization.Team.AddRepository(
+                 contextOrgTeam2.TeamId,
+                repoOwner,
+                repoName,
+                new RepositoryPermissionRequest(Permission.Push));
+
+            var newTeam = new BranchProtectionTeamCollection() { contextOrgTeam2.TeamName };
+            var restrictions = await _client.SetProtectedBranchTeamRestrictions(repoId, "master", newTeam);
+
+            Assert.NotNull(restrictions);
+            Assert.Equal(1, restrictions.Count);
+        }
+
+        public void Dispose()
+        {
+            if (_orgRepoContext != null)
+                _orgRepoContext.Dispose();
+        }
+    }
+
+    public class TheAddProtectedBranchTeamRestrictionsMethod : IDisposable
+    {
+        IRepositoryBranchesClient _client;
+        OrganizationRepositoryWithTeamContext _orgRepoContext;
+        IGitHubClient _github;
+
+        public TheAddProtectedBranchTeamRestrictionsMethod()
+        {
+            _github = Helper.GetAuthenticatedClient();
+            _client = _github.Repository.Branch;
+
+            _orgRepoContext = _github.CreateOrganizationRepositoryWithProtectedBranch().Result;
+        }
+
+        [IntegrationTest]
+        public async Task AddProtectedBranchTeamRestrictionsForOrgRepo()
+        {
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+
+            var contextOrgTeam2 = await _github.CreateTeamContext(Helper.Organization, new NewTeam(Helper.MakeNameWithTimestamp("team2")));
+
+            // Grant team push access to repo
+            await _github.Organization.Team.AddRepository(
+                contextOrgTeam2.TeamId,
+                repoOwner,
+                repoName,
+                new RepositoryPermissionRequest(Permission.Push));
+
+            var newTeam = new BranchProtectionTeamCollection() { contextOrgTeam2.TeamName };
+            var restrictions = await _client.AddProtectedBranchTeamRestrictions(repoOwner, repoName, "master", newTeam);
+
+            Assert.NotNull(restrictions);
+            Assert.Equal(2, restrictions.Count);
+        }
+
+        [IntegrationTest]
+        public async Task AddProtectedBranchTeamRestrictionsForOrgRepoWithRepositoryId()
+        {
+            var repoId = _orgRepoContext.RepositoryContext.RepositoryId;
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+            var contextOrgTeam2 = await _github.CreateTeamContext(Helper.Organization, new NewTeam(Helper.MakeNameWithTimestamp("team2")));
```

Comment:
this context should be disposed when finished so could either put it in a `using ()` block or add it to the class variables like `_orgRepoContext` and add to the class'es `Dispose()` event